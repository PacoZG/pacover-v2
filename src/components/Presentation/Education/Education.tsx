import React from 'react'
import { useTranslations } from 'next-intl'
import '../education-style.css'

const Education: React.FC = () => {
  const t = useTranslations('education')

  return (
    <div className="flex flex-col items-center p-4 md:py-10 lg:py-20">
      <div className="flex flex-col items-center bg-gray-600 dark:bg-gray-900 w-full sm:w-11/12 lg:w-2/3 transition duration-500 rounded-xl bg-opacity-70 p-4">
        <div className="education-card">
          <h1 className="education-card-title">
            <a
              className="education-link"
              href="https://devopswithkubernetes.com/"
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('line12')}
            </a>
          </h1>

          <p className="education-card-description">{t('line11')}</p>
        </div>

        {/* Education Entry 2: DevOps with Docker */}
        <div className="education-card">
          <h1 className="education-card-title">
            <a
              className="education-link"
              href="https://devopswithdocker.com/"
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('line10')}
            </a>
          </h1>

          <p className="education-card-description">{t('line9')}</p>
        </div>

        {/* Education Entry 3: Full Stack Open */}
        <div className="education-card">
          <h1 className="education-card-title">
            <a
              className="education-link"
              href="https://fullstackopen.com/en/"
              target="_blank"
              rel="noopener noreferrer"
            >
              {t('line8')}
            </a>
          </h1>

          <p className="education-card-description">{t('line7')}</p>
        </div>

        {/* Education Entry 4: Aalto University */}
        <div className="education-card">
          <h1 className="education-card-title">{t('line6')}</h1>

          <p className="education-card-description">{t('line5')}</p>

          <a
            education-tool-tip-msg={t('tool-tip-msg')}
            className="education-link-secondary"
            href="https://www.aalto.fi/en/school-of-electrical-engineering"
            target="_blank"
            rel="noopener noreferrer"
          >
            {t('line4')}
          </a>
        </div>

        {/* Education Entry 5: Technical University of Cartagena */}
        <div className="education-card">
          <h1 className="education-card-title">{t('line3')}</h1>
          <p className="education-card-description">{t('line2')}</p>
          <a
            education-tool-tip-msg={t('tool-tip-msg')}
            className="education-link-secondary"
            href="http://www.upct.es"
            target="_blank"
            rel="noopener noreferrer"
          >
            {t('line1')}
          </a>
        </div>
      </div>
    </div>
  )
}

export default Education
