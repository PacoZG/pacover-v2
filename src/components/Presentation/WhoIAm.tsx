import React from 'react'
import { useTranslations } from 'next-intl'

const WhoIAm: React.FC = () => {
  const t = useTranslations('Presentation')

  return (
    <div className="flex justify-center p-3 md:p-10">
      <div className="flex flex-col items-center sm:flex-row justify-center w-full sm:w-11/12 lg:w-2/3 bg-gray-600 dark:bg-gray-900 transition duration-500 bg-opacity-70 dark:bg-opacity-70 rounded-xl p-4">
        <img
          className="h-96 sm:h-80 md:h-96 lg:w-80 border-0 rounded-lg shadow-lg transition duration-500 object-cover"
          src="https://res.cloudinary.com/dbn5gpgi5/image/upload/q_auto/v1653152856/CV%20Pics/idyfixrp3nzidjjezeuc.jpg"
          alt="Profile of Francisco Zavala Gonzalez"
        />
        <div className="p-4 md:p-6 w-full lg:w-3/4 text-left lg:text-left xl:text-left">
          <h1 className="text-3xl text-black dark:text-gray-300 transition duration-500 mb-2 font-extrabold">
            {t('line0')}
            <span className="text-blue-800 dark:text-blue-300 ml-2">
              FRANCISCO ZAVALA GONZALEZ
            </span>
          </h1>

          <p className="text-xl text-black dark:text-gray-300 transition duration-500 mb-4">
            Full Stack Developer
          </p>

          <p className="text-lg text-black dark:text-gray-300 transition duration-500 mb-2 leading-relaxed">
            {t('line1')}
          </p>

          <p className="text-lg text-black dark:text-gray-300 transition duration-500 leading-relaxed">
            {t('line2')}
          </p>
        </div>
      </div>
    </div>
  )
}

export default WhoIAm
