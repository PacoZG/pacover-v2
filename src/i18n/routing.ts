/* istanbul ignore file */
import { defineRouting } from 'next-intl/routing'

export const routing = defineRouting({
  locales: ['en', 'es'],

  defaultLocale: 'en',
})
