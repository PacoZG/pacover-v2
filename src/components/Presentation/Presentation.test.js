import React from 'react'
import { I18nextProvider } from 'react-i18next'
import { BrowserRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import i18n from '../../utils/language/i18next.test'
import Education from './Education.js'
import WhoIAm from './WhoIAm.js'

describe('Testing', () => {
  test('Snapshot of the Education component', () => {
    const redered = renderer
      .create(
        <BrowserRouter>
          <I18nextProvider i18n={i18n}>
            <Education />
          </I18nextProvider>
        </BrowserRouter>
      )
      .toJSON()
    expect(redered).toMatchSnapshot()
  })

  test('Snapshot of the WhoIAm component', () => {
    const redered = renderer
      .create(
        <BrowserRouter>
          <I18nextProvider i18n={i18n}>
            <WhoIAm />
          </I18nextProvider>
        </BrowserRouter>
      )
      .toJSON()
    expect(redered).toMatchSnapshot()
  })
})
