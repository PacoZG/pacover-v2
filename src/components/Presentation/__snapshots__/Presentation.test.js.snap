// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Snapshot of the Education component 1`] = `
<div
  className="bg-gray-400 dark:bg-gray-900 flex flex-col items-center p-4 md:py-10 lg:py-20 transition duration-500 rounded-b-xl bg-opacity-70 dark:bg-opacity-70"
>
  <div
    className="flex flex-col justify-center bg-gray-500 border-2 border-gray-300 rounded-lg p-4 mb-2 w-full sm:w-11/12 lg:w-2/3 sm:h-44 md:h-40 xl:h-36"
  >
    <h1
      className=" md:text-xl text-gray-900 pb-1 font-bold border-b-2 border-gray-300"
    >
      <a
        className="text-gray-900 underline transition duration-200 hover:text-indigo-600 hover:no-underline"
        href="https://devopswithkubernetes.com/"
        target="blank"
      >
        DevOps with Kubernetes (on-line)
      </a>
    </h1>
    <p
      className="text-gray-300 pt-3 text-justify"
    >
      Introductory course to Kubernetes with K3s and GKE lectured on-line in MOOC.fi and developed by the University of Helsinki in partnership with Unity
    </p>
  </div>
  <div
    className="flex flex-col justify-center bg-gray-500 border-2 border-gray-300 rounded-lg p-4 mb-2 w-full sm:w-11/12 lg:w-2/3 sm:h-44 md:h-40 xl:h-36"
  >
    <h1
      className=" md:text-xl text-gray-900 pb-1 font-bold border-b-2 border-gray-300"
    >
      <a
        className="text-gray-900 underline transition duration-200 hover:text-indigo-600 hover:no-underline"
        href="https://devopswithdocker.com/"
        target="blank"
      >
        DevOps with Docker (on-line)
      </a>
    </h1>
    <p
      className="text-gray-300 pt-3 text-justify"
    >
      Introductory course to Docker and docker-compose lectured on-line in MOOC.fi and developed by the University of Helsinki in partnership with Eficode
    </p>
  </div>
  <div
    className="flex flex-col justify-center bg-gray-500 border-2 border-gray-300 rounded-lg p-4 mb-2 w-full sm:w-11/12 lg:w-2/3 sm:h-60 md:h-56 xl:h-48"
  >
    <h1
      className=" md:text-xl text-gray-900 pb-1 font-bold border-b-2 border-gray-300"
    >
      <a
        className="text-gray-900 underline transition duration-200 hover:text-indigo-600 hover:no-underline"
        href="https://fullstackopen.com/en/"
        target="blank"
      >
        Full Stack Open Course (on-line)
      </a>
    </h1>
    <p
      className="text-gray-300 pt-3 text-justify"
    >
      Course lectured on-line in MOOC.fi and developed by the University of Helsinki in cooperation with Partners and affiliates of the course which include Houston Inc, Terveystalo, Elisa, K-ryhmä, Unity Technologies and Konecranes. Completed between AUG 2020 - JULY 2021, written and taught by professors from the University of Helsinki. The course is an introduction to modern web development with JavaScript, React, RESTful and GraphQL web services implemented with Node.js and also have parts on TypeScript, React Native and Continuous integration.
    </p>
  </div>
  <div
    className="flex flex-col justify-center bg-gray-500 border-2 border-gray-300 rounded-lg p-4 mb-2 w-full sm:w-11/12 lg:w-2/3 md:h-36 text-justify"
  >
    <h1
      className=" md:text-xl text-gray-900 pb-1 font-bold border-b-2 border-gray-300"
    >
      Communication Engineering
    </h1>
    <p
      className="text-gray-300  pt-3"
    >
      Master Courses as an Erasmus Exchange Student 2014-2015
    </p>
    <a
      className="text-gray-300  pt-1 underline hover:no-underline transition duration-500"
      education-tool-tip-msg="Click here to visit the school's website"
      href="https://www.aalto.fi/en/school-of-electrical-engineering"
      target="blank"
    >
      @Aalto University – School of Electrical Engineering
    </a>
  </div>
  <div
    className="flex flex-col justify-center bg-gray-500 text-gray-900 border-2 border-gray-300 rounded-lg p-4 w-full sm:w-11/12 lg:w-2/3 md:h-36"
  >
    <h1
      className=" md:text-xl text-gray-900 pb-1 font-bold border-b-2 border-gray-300"
    >
      Bachelor in Telecommunications Engineering
    </h1>
    <p
      className="text-gray-300 pt-3"
    >
      I reached the 4th academic year with 78% of the studies completed
    </p>
    <a
      className="text-gray-300 pt-1 underline hover:no-underline transition duration-500"
      education-tool-tip-msg="Click here to visit the school's website"
      href="http://www.upct.es"
      target="blank"
    >
      @Universidad Politecnica de Cartagena - Technical School of Telecommunications Engineering
    </a>
  </div>
</div>
`;

exports[`Testing Snapshot of the WhoIAm component 1`] = `
<div
  className="flex justify-center bg-gray-400 dark:bg-gray-900 p-3 md:p-10 w-full transition duration-500 bg-opacity-70 dark:bg-opacity-70"
>
  <div
    className="flex flex-col items-center sm:flex-row justify-center sm:w-11/12 lg:w-9/12 "
  >
    <img
      alt="profile"
      className=" h-96 sm:h-80 md:h-96 lg:w-80 border-0 rounded-lg transition duration-500 "
      src="https://res.cloudinary.com/dbn5gpgi5/image/upload/q_auto/v1653152856/CV%20Pics/idyfixrp3nzidjjezeuc.jpg"
    />
    <div
      className="p-10 md:p-12 md:pr-0 md:pt-8 w-full lg:w-3/4"
    >
      <h1
        className="text-2xl text-black dark:text-gray-300 text-right transition duration-500"
      >
        HI! I AM 
        <span
          className="text-blue-800 dark:text-blue-300 text-right text-2xl transition duration-500"
        >
          FRANCISCO ZAVALA GONZALEZ
        </span>
      </h1>
      <p
        className="text-black dark:text-gray-300 text-right text-xl transition duration-500"
      >
        Full Stack Developer
      </p>
       
      <br />
      <p
        className="text-lg text-black dark:text-gray-300 text-right transition duration-500"
      >
        I am Mexican, I have been living in Finland since September 2014
      </p>
      <p
        className="text-lg text-black dark:text-gray-300 text-right transition duration-500"
      >
        I started my studies in Communication Engineering in Spain. In the year 2014 I moved to Finland to study some master courses at Aalto University as an Erasmus student.
      </p>
    </div>
  </div>
</div>
`;
